spec/controllers/stages_controller_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe StagesController, type: :controller do
+  before do
+    @stage = create(:playoff_stage)
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      get :show, params: { id: @stage.to_param }
+      expect(response).to be_successful
+    end
+
+    it 'should return the correct stage' do
+      get :show, params: { id: @stage.to_param }
+      body = deserialize_response response
+      expect(Stage.find_by(id: body[:id])).to eq(@stage)
+      expect(body[:level]).to eq(@stage.level)
+      expect(body[:state]).to eq(@stage.state)
+    end
+  end
+
+  describe 'PUT #update' do
+    context 'group_stage with matches that are done' do
+      before do
+        @running_group_stage = create(:group_stage, match_factory: :finished_group_match)
+      end
+
+      FINISHED = { state: 'finished' }.freeze
```

Comment:
why is this a constant instead of a let statement?

Reply:
because random.org